### DISCUSSION TITLE
Email Webhook Sync on gmail 📨

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#3937] [Ideas]
As of today the email sync is done through a cron that runs every 10 minutes.
We'll need to provide immediate sync through webhooks.

We'll start with gmail.

- Create an environment variables to store the topic. Document it in `enviroment-variables.mdx`. Default value = gmail
- Call the /watch endpoint to subscribe to account upon mailbox connection 
- Create an endpoint to receive subscriptions events and import email ([ref](https://developers.google.com/gmail/api/guides/push#getting_mailbox_updates))
- Create the [topic](https://cloud.google.com/pubsub/docs/create-topic#pubsub_create_topic-Console
) and [push subscription](https://cloud.google.com/pubsub/docs/subscriber) in our Google Cloud Console  
- Setup a cron to renew the /watch API call, configure it to run daily

**Comment by bonnlei** [comment]
How to setup AUTH_GOOGLE_CALLBACK_URL? Is it http://localhost:3000/auth/verify? Why do I have "Missing required parameter: redirect_uri by google auth?" error by sync gmail? Where is enviroment-variables.mdx?

**Reply by FelixMalfait** [reply]
It's been moved to self-hosting.mdx (please raise a PR if there are outdated references that would help, thanks)
And the `.env.example` value is `http://localhost:3000/auth/google/redirect` (replace localhost:3000 by your domain)

**Reply by bonnlei** [reply]
thx

**Comment by BOHEUS** [comment]
Created issue https://github.com/twentyhq/twenty/issues/18790 based on discussion

**Reply by FelixMalfait** [reply]
@BOHEUS not sure why? It's a big workstream (potentially weeks to implement it cleanly), better as a discussion for future roadmap no?

**Reply by BOHEUS** [reply]
I was thinking it's feasible in short-term (1 sprint = 2 weeks) to add a webhook in codebase which forwards the received webhook to standard flow of adding message to Twenty and extracting participants but maybe I'm missing something there

**Reply by neo773** [reply]
This would be done as part of a larger work to make it work across all drivers with abstraction
I did some preliminary research and work on it a while ago
